### PR TITLE
Update

### PR DESCRIPTION
**Pull request by rramphal**
In hopes of updating this great module after two years, I:
- updated the dictionary dump text file
- regenerated the SQLite database
- updated dependencies (and made according changes to the code) including Sequelize and its promises
- added JSHint config file
- switched from the `assert` library to the `chai` library (`assert`'s development seems to be faulty as of late)
- made whitespace consistent

Once this is merged in, I hope to add the `searchByEnglish` functionality.

### CODE REVIEW

**Comment by evansque** [comment]
Fantastic! this should be merged soon.
note: **CC-CEDICT Latest release: 2017-03-03 21:33:39 GMT**

File: README.md [hunk 1]
```diff
@@ -9,12 +9,12 @@ Single search method defaults to traditional if the entire word provided is in t
 ```
 var cedict = require('node-cc-cedict');
 
-cedict.searchByChinese('世界', function(words){
+cedict.searchByChinese('世界', function (words) {
```
Comment (johnheroy) [inline README.md]: Please don't change the overall JS style of the library

File: index.js [hunk 2]
```diff
@@ -1,100 +1,122 @@
...
 var _ = require('underscore');
+var cnchars = require('cn-chars'); // convert between traditional and simplified
 var path = require('path');
+var pinyin = require('prettify-pinyin'); // prettify the pinyin default (letters + numbers) in CC-CEDICT
 var Sequelize = require('sequelize');
+
 var sequelize = new Sequelize(null, null, null, {
   dialect: 'sqlite',
-  storage: path.join(__dirname, './db/cc-cedict.sqlite')
+  storage: path.join(__dirname, './db/', 'cc-cedict.sqlite')
```
Comment (johnheroy) [inline index.js:11]: Why is this changed?

File: index.js [hunk 3]
```diff
@@ -1,100 +1,122 @@
...
 });
 
-var Word = sequelize.define('Word', {
```
Comment (johnheroy) [inline index.js:9]: What is the context for these changes?

File: index.js [hunk 4]
```diff
@@ -1,100 +1,122 @@
...
+  }
 });
 
-// convert between traditional and simplified
-var cnchars = require('cn-chars');
-// prettify the pinyin default (letters + numbers) in CC-CEDICT
-var pinyin = require('prettify-pinyin');
-
-module.exports.searchByChinese = function(str, cb){
+module.exports.searchByChinese = function (str, cb) {
```
Comment (johnheroy) [inline index.js]: There should not be any whitespace in between the function keyword and opening parens, as well as after the closing parens after parameters, e.g.:

function(a, b){
  // do stuff
}

File: parser/dictionary-parser.js [hunk 5]
```diff
@@ -1,48 +1,55 @@
...
 var fs = require('fs');
+var path = require("path");
+var Sequelize = require('sequelize');
 
-// defined db config
+// define db config
 var sequelize = new Sequelize(null, null, null, {
   dialect: 'sqlite',
-  storage: '../db/cc-cedict.sqlite'
+  storage: path.join(__dirname, '../db/', 'cc-cedict.sqlite')
```
Comment (johnheroy) [inline parser/dictionary-parser.js:14]: What's the rationale for this change?

File: README.md (hunk 1 above)
Comment (rramphal) [reply README.md]: Restored in 84807a51e57065cb987fcb719b4351d9ceb467ad.

File: index.js (hunk 2 above)
Comment (rramphal) [reply index.js:11]: This is best practice.

File: index.js (hunk 3 above)
Comment (rramphal) [reply index.js:9]: MySQL convention does not have capitalized table names (which this line generates).

File: index.js (hunk 4 above)
Comment (rramphal) [reply index.js]: Restored in 84807a51e57065cb987fcb719b4351d9ceb467ad.

File: parser/dictionary-parser.js (hunk 5 above)
Comment (rramphal) [reply parser/dictionary-parser.js:14]: This is best practice.

**Comment by rramphal** [comment]
Hey @johnheroy, are there any more changes that you'd like me to make or any more questions that you have before this can be merged?

**Comment by johnheroy** [comment]
I am not going to have the bandwidth to review this thoroughly and would suggest just working off of your own fork if your project depends on this node module.
